src/linux/system.rs

```diff
@@ -786,24 +802,42 @@ enum InfoType {
 }
 
 #[cfg(not(target_os = "android"))]
-fn get_system_info(info: InfoType) -> Option<String> {
+fn get_system_info<T>(info: InfoType, buf_read: Option<T>) -> Option<String>
+where
+    T: BufRead,
+{
     let info_str = match info {
         InfoType::Name => "NAME=",
         InfoType::OsVersion => "VERSION_ID=",
     };
 
-    let buf = BufReader::new(File::open("/etc/os-release").ok()?);
+    if let Some(reader) = buf_read {
+        for line in reader.lines().flatten() {
+            if let Some(stripped) = line.strip_prefix(info_str) {
+                return Some(stripped.replace("\"", ""));
+            }
+        }
+    }
 
-    for line in buf.lines().flatten() {
-        if let Some(stripped) = line.strip_prefix(info_str) {
-            return Some(stripped.replace("\"", ""));
+    // Fallback to `lsb_release -r` for systems where VERSION_ID is not included.
+    // VERSION_ID is no required per https://www.linux.org/docs/man5/os-release.html
+    // If this fails for some reason, fallback to None
+    if info == InfoType::OsVersion {
+        if let Ok(lsb_out) = Command::new("lsb_release").args(&["-r"]).output() {
```

Comment:
Please don't run executables. Please take a look at how they do but no command should be started.

Reply:
Looks like there is an additional file to read from, so I'll swap over to that.  Just wondering why the no executable?

Reply:
Personal opinion mostly, but:
 * Binaries versions can change a lot between two computers (and therefore their output too)
 * It's quite slow to call another binary
 * The binary might not be present
 * The binary could fail and providing a nice error would be very tricky